engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/compile/ProcessCompiler.scala

```diff
@@ -126,12 +129,23 @@ protected trait ProcessCompilerBase {
     CompilationResult.map3(
       CompilationResult(findDuplicates(splittedProcess.sources).toValidatedNel),
       CompilationResult(nodeTypingInfo, compiledExceptionHandler),
-      splittedProcess.sources.map(compile).sequence
+      compileSources(splittedProcess.sources)
     ) { (_, exceptionHandler, sources) =>
       CompiledProcessParts(splittedProcess.metaData, exceptionHandler, sources)
     }
   }
 
+  private def compileSources(sources: NonEmptyList[SourcePart])(implicit meta: MetaData): CompilationResult[NonEmptyList[PotentiallyStartPart]] = {
+    val zeroAcc = (CompilationResult(Valid(List[PotentiallyStartPart]())), BranchEndContexts(Map(), None))
+    val (result, _) = PartSort.sort(sources.toList).foldLeft(zeroAcc) { case ((resultSoFar, branchContexts), nextPart) =>
+        val compiledPart = compile(nextPart, branchContexts)
+        //we don't use andThen on CompilationResult, since we don't want to stop if there are errors in part
+        val nextResult = CompilationResult.map2(resultSoFar, compiledPart)(_ :+ _)
+        (nextResult, branchContexts.addPart(nextPart.node, compiledPart))
```

Comment:
I don't get it ... we add part also for non-join parts? Do we really need to do it in folding (next step depends on previous) or we can do it without such dependency: first some map and then sequence

Reply:
I added comment in code